ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -757,8 +763,15 @@ private static ComparisonTarget extractWhereClauseTarget(
   }
 
   private static boolean isSelectStar(final Select select) {
-    final List<SelectItem> selects = select.getSelectItems();
-    return selects.size() == 1 && selects.get(0) instanceof AllColumns;
+    final boolean someStars = select.getSelectItems().stream()
+        .anyMatch(s -> s instanceof AllColumns);
+
+    if (someStars && select.getSelectItems().size() != 1) {
+      throw new KsqlException("Pull queries only support wildcards in the projects "
+          + "if they are the only expression");
+    }
```

Comment:
This actually means KSQL rejects any pull query that uses a mix of `*` and explicit column names in a pull query. Previously, depending on the order, the explicit columns were ignored or a weird error was returned.  At least now a meaningful error is returned.